aip/general/0216.md

```diff
@@ -123,6 +123,44 @@ message PublishBookRequest {
 - The comment for the field **should** document the resource pattern.
 - Other fields **may** be included.
 
+### Declarative-friendly state transitions
+
+If a resource's state can be modified by [declarative tooling][aip-128], it
+**should** expose a separate `desired_state` field taking the same enum type.
+
+```proto
+message Book {
+  enum State {
+    STATE_UNSPECIFIED = 0;
+    UNPUBLISHED = 1;
+    PUBLISHING = 2;
+    PUBLISHED = 3;
+  }
+  
+  State state = 1 [(google.api.field_behaviopr = OUTPUT_ONLY)];
+  State desired_state = 2;
+}
+```
+
+- The `state` field's output **may** change to more granular
+  [active states](#common-states) to indicate progress toward the `desired_state`.
+- The `desired_state` field **must** only allow [resting state](#common-states)
```

Comment:
As discussed above, the best way to enforce this is to use a separate enum entirely. 

Doing so eliminates the need to require the runtime behavior outlined below (preconditions, invalid_argument)